src/participant/participant.ts

```diff
@@ -0,0 +1,332 @@
+  }): Promise<string> {
+    let responseContent = '';
+    try {
+      const [model] = await vscode.lm.selectChatModels({
+        vendor: 'copilot',
+        family: CHAT_PARTICIPANT_MODEL,
+      });
+      if (model) {
+        const chatResponse = await model.sendRequest(messages, {}, token);
+        for await (const fragment of chatResponse.text) {
+          responseContent += fragment;
+          stream.markdown(fragment);
+        }
+        stream.markdown('\n\n');
+      }
+    } catch (err) {
+      this.handleError(err, stream);
+    }
+
+    return responseContent;
+  }
+
+  // @MongoDB what is mongodb?
+  async handleGenericRequest({
+    request,
+    context,
+    stream,
+    token,
+  }: {
+    request: vscode.ChatRequest;
+    context: vscode.ChatContext;
+    stream: vscode.ChatResponseStream;
+    token: vscode.CancellationToken;
+  }) {
+    const messages = [
+      // eslint-disable-next-line new-cap
+      vscode.LanguageModelChatMessage.Assistant(`You are a MongoDB expert!
+  You create MongoDB queries and aggregation pipelines,
+  and you are very good at it. The user will provide the basis for the query.
+  Keep your response concise. Respond with markdown, code snippets are possible with '''javascript.
+  You can imagine the schema, collection, and database name.
+  Respond in MongoDB shell syntax using the '''javascript code style.'.`),
+    ];
+
+    context.history.map((historyItem) => {
+      if (
+        historyItem.participant === CHAT_PARTICIPANT_ID &&
+        historyItem instanceof vscode.ChatRequestTurn
+      ) {
+        // eslint-disable-next-line new-cap
+        messages.push(vscode.LanguageModelChatMessage.User(historyItem.prompt));
+      }
+
+      if (
+        historyItem.participant === CHAT_PARTICIPANT_ID &&
+        historyItem instanceof vscode.ChatResponseTurn
+      ) {
+        let res = '';
+        for (const fragment of historyItem.response) {
+          res += fragment;
+        }
+        // eslint-disable-next-line new-cap
+        messages.push(vscode.LanguageModelChatMessage.Assistant(res));
+      }
+    });
+
+    // eslint-disable-next-line new-cap
+    messages.push(vscode.LanguageModelChatMessage.User(request.prompt));
+
+    const abortController = new AbortController();
+    token.onCancellationRequested(() => {
+      abortController.abort();
+    });
+
+    const responseContent = await this.getChatResponseContent({
+      messages,
+      stream,
+      token,
+    });
+    const queryContent = getRunnableContentFromString(responseContent);
+
+    if (!queryContent || queryContent.trim().length === 0) {
+      return { metadata: { command: '' } };
+    }
+
+    stream.button({
+      command: EXTENSION_COMMANDS.RUN_PARTICIPANT_QUERY,
+      title: vscode.l10n.t('▶️ Run'),
+    });
+
+    stream.button({
+      command: EXTENSION_COMMANDS.OPEN_PARTICIPANT_QUERY_IN_PLAYGROUND,
+      title: vscode.l10n.t('Open in playground'),
+    });
```

Comment:
Maybe not for this pr, but I think there is a chance the ai could return multiple code blocks in their response. 
Like the prompt:
> What are 3 ways I could do X

I'm thinking we probably want to add a play and open in playground button for each code block as they come. Ideally we show them inline, right after the code block closes, while we stream the response.

Reply:
Good point. Let's address this in further PRs 👍